src/sentry/workflow_engine/handlers/condition/event_frequency_handlers.py

```diff
@@ -71,16 +71,14 @@ class EventFrequencyCountHandler(EventFrequencyConditionHandler, DataConditionHa
     }
 
     @staticmethod
-    def evaluate_value(value: WorkflowJob, comparison: Any) -> DataConditionResult:
-        if len(value.get("snuba_results", [])) != 1:
+    def evaluate_value(value: DataJob, comparison: Any) -> DataConditionResult:
+        if not value.get("results") or len(value.get("results", [])) != 1:
```

Comment:
NOTE: when evaluating all conditions a `WorkflowJob` will be passed to slow conditions, which actually take in `DataJob`. i couldn't find a way around this to make mypy happy, but since `results` is optional on `DataJob` i check for its existence which conveniently accounts for `WorkflowJob` not having results in the first place...